sympy/vector/coordsysrect.py

```diff
@@ -213,6 +214,37 @@ def _connect_to_standard_cartesian(self,  curv_coord_type):
         else:
             raise ValueError("Wrong set of parameter.")
 
+        if not self._check_orthogonality():
+            raise ValueError("The transformation equation does not create orthogonal coordinate system")
+
+    def _check_orthogonality(self):
+        """
+        Helper method for _connect_to_cartesian. It checks if
+        set of transformation equations create orthogonal curvilinear
+        coordinate system
+
+        Parameters
+        ==========
+
+        equations : tuple
+            Tuple of transformation equations
+
+        """
+
+        eq = self._transformation_equations()
+
+        v1 = Matrix([diff(eq[0], self.x), diff(eq[1], self.x), diff(eq[2], self.x)])
+        v2 = Matrix([diff(eq[0], self.y), diff(eq[1], self.y), diff(eq[2], self.y)])
+        v3 = Matrix([diff(eq[0], self.z), diff(eq[1], self.z), diff(eq[2], self.z)])
+
+        if any(simplify(i[0]+i[1]+i[2]) == 0 for i in (v1, v2, v3)):
```

Comment:
This condition is technically the condition of non-degenerancy, it is not related to the orthogonality.